src/SignedSettlement.sol

```diff
@@ -0,0 +1,82 @@
+pragma solidity ^0.8;
+
+import {Auth} from "./Auth.sol";
+import {GPv2Trade} from "cowprotocol/libraries/GPv2Trade.sol";
+import {GPv2Interaction} from "cowprotocol/libraries/GPv2Interaction.sol";
+import {LibSignedSettlement} from "./LibSignedSettlement.sol";
+import {ECDSA} from "solady/utils/ECDSA.sol";
+import {SubPoolFactory} from "./SubPoolFactory.sol";
+import {GPv2Settlement, IERC20} from "cowprotocol/GPv2Settlement.sol";
+
+contract SignedSettlement is Auth {
+    error SignedSettlement__InvalidAttestor();
+    error SignedSettlement__DeadlineElapsed();
+
+    /// @notice whether signer can attest
+    address public immutable attestor;
+    GPv2Settlement public immutable settlement;
+
+    constructor(GPv2Settlement _settlement, address _attestor) {
+        settlement = _settlement;
+        attestor = _attestor;
+    }
+
+    /// @notice Takes the required settlement data and verifies that it has been
+    ///         signed by a `signer`, and subsequently calls `GPv2Settlement.settle`.
```

Comment:
Would somehow add here to the docs that there are hidden parameters expected when being called.

Also do we have an idea how gas expensive it would be to make these values explicit (because now you kind of need a Web3 degree in order to understand what's going on here and we rely a lot on assembly which can be a footgun)

Reply:
It's possible, just that you will have to shift all the offsets out

Reply:
added docs